volatility3/framework/plugins/regexscan.py

```diff
@@ -46,38 +46,63 @@ def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]
             ),
         ]
 
-    def _generator(self, regex_pattern):
-        regex_pattern = bytes(regex_pattern, "UTF-8")
-        vollog.debug(f"RegEx Pattern: {regex_pattern}")
-
+    def _generator(self, compiled_pattern, raw_pattern, maxsize):
+        vollog.debug(f"RegEx Pattern: {raw_pattern}")
         layer = self.context.layers[self.config["primary"]]
+
         for offset in layer.scan(
-            context=self.context, scanner=scanners.RegExScanner(regex_pattern)
+            context=self.context, scanner=scanners.RegExScanner(raw_pattern)
         ):
-            result_data = layer.read(offset, self.MAXSIZE_DEFAULT, pad=True)
+            result_data = layer.read(offset, maxsize, pad=True)
 
             # reapply the regex in order to extract just the match
-            regex_result = re.match(regex_pattern, result_data)
+            regex_result = compiled_pattern.search(result_data)
 
             if regex_result:
-                # the match is within the results_data (e.g. it fits within MAXSIZE_DEFAULT)
+                # the match is within the results_data (e.g. it fits within maxsize)
                 # extract just the match itself
                 regex_match = regex_result.group(0)
                 text_result = str(regex_match, encoding="UTF-8", errors="replace")
                 bytes_result = regex_match
             else:
-                # the match is not with the results_data (e.g. it doesn't fit within MAXSIZE_DEFAULT)
+                # the match is not with the results_data (e.g. it doesn't fit within maxsize)
                 text_result = str(result_data, encoding="UTF-8", errors="replace")
                 bytes_result = result_data
 
             yield 0, (format_hints.Hex(offset), text_result, bytes_result)
 
     def run(self):
+        pattern = self.config.get("pattern")
+
+        # Handle pattern encoding robustly
+        if isinstance(pattern, str):
+            try:
+                raw_pattern = pattern.encode("utf-8")
+            except UnicodeEncodeError:
+                raw_pattern = pattern.encode("latin1", errors="replace")
+        else:
+            raw_pattern = pattern
+
+        try:
+            compiled_pattern = re.compile(raw_pattern)
+        except re.error as e:
+            vollog.error(f"Invalid regex pattern: {e}")
+            return renderers.TreeGrid(
```

Comment:
This probably wants to be de-indented, don't throw an error but then return a tree grid (even an empty one).  If it's thrown an error, stop the program or raise an exception or *something*, don't carry on like everything's fine.